adapters/sovrnXsp/sovrnXsp.go

```diff
@@ -0,0 +1,172 @@
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("imp #%d: ext.bidder not provided", idx),
+			}
+			errors = append(errors, err)
+			continue
+		}
+
+		var xspExt openrtb_ext.ExtImpSovrnXsp
+		if err := json.Unmarshal(bidderExt.Bidder, &xspExt); err != nil {
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("imp #%d: %s", idx, err.Error()),
+			}
+			errors = append(errors, err)
+			continue
+		}
+
+		request.App.Publisher.ID = xspExt.PubID
+		if xspExt.MedID != "" {
+			request.App.ID = xspExt.MedID
+		}
+		if xspExt.ZoneID != "" {
+			imp.TagID = xspExt.ZoneID
+		}
+		imps = append(imps, imp)
+	}
+
+	if len(imps) == 0 {
+		return nil, append(errors, &errortypes.BadInput{
+			Message: "no matching impression with ad format",
+		})
+	}
+
+	request.Imp = imps
+	requestJson, err := json.Marshal(request)
+	if err != nil {
+		return nil, append(errors, err)
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     a.Endpoint,
+		Body:    requestJson,
+		Headers: headers,
+	}}, errors
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if adapters.IsResponseStatusCodeNoContent(responseData) {
+		return nil, nil
+	}
+	if err := adapters.CheckResponseStatusCodeForErrors(responseData); err != nil {
+		return nil, []error{err}
+	}
+
+	var response openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	var errors []error
+	result := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+
+	for _, seatBid := range response.SeatBid {
+		for _, bid := range seatBid.Bid {
+			bid := bid
+			var ext bidExt
+			if err := json.Unmarshal(bid.Ext, &ext); err != nil {
+				errors = append(errors, err)
+				continue
+			}
+
+			var bidType openrtb_ext.BidType
+			var mkupType openrtb2.MarkupType
+			switch ext.CreativeType {
+			case creativeTypeBanner:
+				bidType = openrtb_ext.BidTypeBanner
+				mkupType = openrtb2.MarkupBanner
+			case creativeTypeVideo:
+				bidType = openrtb_ext.BidTypeVideo
+				mkupType = openrtb2.MarkupVideo
+			case creativeTypeNative:
+				bidType = openrtb_ext.BidTypeNative
+				mkupType = openrtb2.MarkupNative
+			default:
+				errors = append(errors, &errortypes.BadServerResponse{
+					Message: fmt.Sprintf("Unsupported creative type: %d", ext.CreativeType),
+				})
+				continue
+			}
+
+			if bid.MType == 0 {
+				bid.MType = mkupType
+			}
+
+			result.Bids = append(result.Bids, &adapters.TypedBid{
+				Bid:     &bid,
+				BidType: bidType,
+			})
+		}
+	}
+
+	if len(result.Bids) == 0 {
+		// it's possible an empty seat array was sent as a response
+		return nil, errors
```

Comment:
should append empty seat array in error in errors

```
if len(result.Bids) == 0 {
    // it's possible an empty seat array was sent as a response
    - return nil, errors
    + return nil, append(errors, errors.New("empty seat bit array"))
}
```

Reply:
We don't consider an empty seat array as an error. It's a different codepath that signals no-bid.